compiler-plugin/src/main/scala/com/trueaccord/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -1074,6 +1074,7 @@ class ProtobufGenerator(val params: GeneratorParams) extends DescriptorPimps {
       s"""object $className extends $companionType {
          |  implicit def messageCompanion: $companionType = this""")
       .indent
+      .add(s"type ValueType = $className")
```

Comment:
No, the difference is between adding this type to each individual instance versus adding it just once to the base trait. The latter has the advantage that generic code that works with the base trait will be able to access ValueType. Does that make sense?